jetty-websocket/javax-websocket-server/src/main/java/org/eclipse/jetty/websocket/javax/server/JavaxWebSocketServerContainer.java

```diff
@@ -88,26 +87,39 @@ public static JavaxWebSocketServerContainer ensureContainer(ServletContext servl
         JavaxWebSocketServerContainer container = contextHandler.getBean(JavaxWebSocketServerContainer.class);
         if (container==null)
         {
-            // Find Pre-Existing (Shared?) HttpClient and/or executor
-            HttpClient httpClient = (HttpClient)servletContext.getAttribute(JavaxWebSocketServletContainerInitializer.HTTPCLIENT_ATTRIBUTE);
-            if (httpClient == null)
-                httpClient = (HttpClient)contextHandler.getServer()
-                    .getAttribute(JavaxWebSocketServletContainerInitializer.HTTPCLIENT_ATTRIBUTE);
-
-            Executor executor = httpClient == null?null:httpClient.getExecutor();
-            if (executor == null)
-                executor = (Executor)servletContext.getAttribute("org.eclipse.jetty.server.Executor");
-            if (executor == null)
-                executor = contextHandler.getServer().getThreadPool();
-
-            if (httpClient != null && httpClient.getExecutor() == null)
-                httpClient.setExecutor(executor);
+            Supplier<WebSocketCoreClient> coreClientSupplier = () ->
+            {
+                WebSocketCoreClient coreClient = contextHandler.getBean(WebSocketCoreClient.class);
```

Comment:
I know it was my suggestion to start using beans instead of attributes... but now I'm not so sure.  Why is the client an attribute, but the cloreClient a bean?    Do we have a good reason why some things are beans and others are attributes?   If not, let's go back to all attributes.